dashboard/src/components/AppList/AppList.tsx

```diff
@@ -73,6 +75,12 @@ function AppList() {
     dispatch(actions.operators.getResources(cluster, namespace));
   }, [dispatch, cluster, namespace]);
 
+  useEffect(() => {
+    // In order to be able to list applications in all namespaces, it's necessary to be able
+    // to list/get secrets in all of them.
+    Kube.canI(cluster, "", "secrets", "list", "").then(allowed => setCanSetAllNS(allowed));
```

Comment:
I'm still not sure why having a single "userConfig" check when the app loads to get back the user config and store that in state wouldn't be better than using `canI` wherever the UI wants to make a decision. The UI here wants to know if the user can view apps across all namespaces, but the UI has to jump down to lower level implementation of how to determine that - listing secrets across namespaces. Our backend API could encapsulate those details and just provide the user config (what they can and can't do).

Reply:
IMO, this is not "userConfig" since we are just talking about permissions (for me "config" would mean something that the user can change and configure at will). So if we want to move this to a single call in the backend it would be "userPermissions" (since at the moment, the backend has nothing configurable as far as I can remember).

Now, the "problems" with that are two (again, in my opinion):
 - It's slower to implement. I would need to implement things in two places, write tests in two places, modify the API and build the image. 
 - The response wouldn't be structured. If someone takes a look to the backend standalone, it wouldn't make sense to see why we are returning only certain permissions (create namespaces, list secrets, list apprepositories) while it's only necessary in a very specific point of the UI.

On the other hand, it's true that here the frontend needs to know that the permission to list apps maps to the permission to list secrets (and that's a decision made in the backend). Not sure if avoiding that assumption in the UI is worth the effort/downsides of the other approach.

I will merge this as it is in the dev branch but we can keep the discussion and I can open a different PR.

Reply:
Sure, `userConfig` could be `userPerms`, that's just naming. Regarding the other point about structure, yes - it'd be encapsulating all the info that the UI needs about the user to operate - totally agree it'd be slightly more work but for the benefit of encapsulating that info in the one call/place.

> I will merge this as it is in the dev branch but we can keep the discussion and I can open a different PR.

Yep, i don't feel strongly, just worth discussion.